bash/users.sh

```diff
@@ -0,0 +1,263 @@
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+# DEALINGS IN THE SOFTWARE.
+
+echo "Example script using the user's API"
+echo
+# Check cURL command if available (required), abort if does not exists
+type curl >/dev/null 2>&1 || { echo >&2 "Required curl but it's not installed. Aborting."; exit 1; }
+echo
+
+# Make sure Variables are encapsulated with "" if there are spaces within them
+uuid='<Your UUID>'
+email='<Your admin or first user email>'
+password='<Your Password>'
+
+# These are some of the endpoints of the Lyve Pilot API that can be queried.
+# This script is written to use all url endpoints as global variables and used by
+# helper functioins. 
+USERS_URL='/udx/v1/users'
+NEW_USER_URL='/udx/v1/saasUser'
+ACCEPT_EULA_URL='/udx/v1/acceptEULAByUser'
+SET_PASSWORD_URL='/udx/v1/setSaasPassword'
+FEED_URL='/udx/v1/feed'
+
+BASE_URL='https://api-staging.lyve.seagate.com/'$uuid
+
+name="John Doe"
+
+# There are two ways to do JSON objects in bash using '' will make it impossible to
+# use variables in the string. Instead use "" and escape the extra "" to create a 
+# valid JSON string
+new_user_data='{"email":"email@organizaton.com","role":"admin","language":"EN","location":"0","name":"John Doe","phone":"123-456-7890"}'
+
+#  login(username, password) - This currently logs the user in and 
+#          returns the entire JSON object which includes the session JWT and a refresh token.
+#          The refresh token can be used to refresh the JWT when it expires.
+#  Input - username: username  $1
+#          password: password  $2
+#  Output - Returns a json web token that is passed through other functions
+#           as a valiidation object. Expires after 10 minutes. Also returns
+#           a refresh token in order to extend session
+
+login () {
+    local data="{\"email\": \"$1\",\"password\": \"$2\"}"
+    response=`curl -s --request POST -H "Content-type:application/json" $BASE_URL/udx/v1/login --data "$data"`
+    # One way of parsing JSON by using grep
+    token=`echo $response | grep -o '"token":"[^"]*' | grep -o '[^"]*$'`
+    echo $token
+}
+
+#  get_users(headers, url) - Fetches an array of usernames currently
+#                         saved to customer instance as strings
+#  Input - headers: the headers for the curl request  $1
+#          url: Desired url endpoint  $2
+#  Output - Returns a string of JSON objects in the form of {name: name}
+get_users(){
+    response=$(get_helper "$1" $2)
+    # another way of parsing JSON by using jq
+    names=`$response | jq '.["data"] | .[] | {name: .name}'`
+    echo $names
+}
+
+# id_from_username(headers, url, username) - Since user ID's are long and
+#                                         randomly generated, users are
+#                                         not expected to remember them.
+#                                         This function finds the ID from
+#                                         username
```

Comment:
Is there a better way to format this? Seems like it is hard to get the indentation right as we are squishing the text to the right.

I like explicitly showing the arguments. Maybe there is a way to do that while avoiding the description squish.